ee/agent/keys_tpm.go

```diff
@@ -28,17 +31,24 @@ func setupHardwareKeys(slogger *slog.Logger, store types.GetterSetterDeleter) (k
 		priData, pubData, err = tpm.CreateKey()
 		if err != nil {
 			clearKeyData(slogger, store)
+			traces.SetError(span, fmt.Errorf("creating key: %w", err))
 			return nil, fmt.Errorf("creating key: %w", err)
 		}
 
+		span.AddEvent("new_key_created")
```

Comment:
do we need this? It's somewhat implied by the function name. I guess we're still learning the tools

Reply:
I like seeing the events when I look at a span, but I can live without them.